apps/server/src/modules/authorization-rules/rules/instance.rule.ts

```diff
@@ -25,16 +25,46 @@ export class InstanceRule implements Rule<Instance> {
 		return isMatched;
 	}
 
-	public hasPermission(user: User, entity: Instance, context: AuthorizationContext): boolean {
-		const hasPermission = this.authorizationHelper.hasAllPermissions(user, context.requiredPermissions);
+	public hasPermission(user: User, instance: Instance, context: AuthorizationContext): boolean {
+		let hasPermission = false;
 
-		// As temporary solution until the user with write access to instance added as group, we must check the role.
+		if (context.action === Action.read) {
+			hasPermission = this.hasReadAccess(user, instance, context);
+		}
 		if (context.action === Action.write) {
-			const hasRole = this.authorizationHelper.hasRole(user, RoleName.SUPERHERO);
-
-			return hasPermission && hasRole;
+			hasPermission = this.hasWriteAccess(user, instance, context);
 		}
 
 		return hasPermission;
 	}
+
+	private hasReadAccess(user: User, instance: Instance, context: AuthorizationContext): boolean {
+		/**
+		 * @notImplemented Currently we do not have user relations in instance.
+		 */
+		const hasReadPermission = false;
```

Comment:
Wieso dann nicht weglassen, wenn es nicht gebraucht wird?

Reply:
Um auszudrücken das hier noch etwas fehlt. Wir haben bei allen anderen Rules moment ein user context, selbst impliziet bei Schule über user.schoolId + Role. Nur hier wurde er bisher nicht umgesetzt.

Reply:
Aber was soll der Leser damit anfangen? Dass sie nicht da sind, sieht er ja selber. MMn sollte der Kommentar wenn dann sagen, WARUM sie nicht da sind und was das bedeutet. 

Und was heißt eigentlich diese Schreibweise mit `@notImplemented`? Wird das geparset für irgendwas? Ich sehe das sonst nicht in unserem Code.

Reply:
Welche Tests meinst du da genau?

Reply:
done